imap_processing/tests/pointing_frame/test_pointing_frame_handler.py

```diff
@@ -0,0 +1,136 @@
+"""Tests Pointing Frame Generation."""
+
+import shutil
+import sys
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pytest
+import spiceypy as spice
+
+from imap_processing.pointing_frame_handler import (
+    average_quaternions,
+    create_pointing_frame,
+    create_rotation_matrix,
+    get_coverage,
+)
+
+
+@pytest.fixture()
+def kernel_path(tmp_path):
+    """Create path to kernels."""
+
+    test_dir = (
+        Path(sys.modules[__name__.split(".")[0]].__file__).parent
+        / "tests"
+        / "pointing_frame"
+        / "test_data"
+    )
+
+    kernels = [
+        "de430.bsp",
+        "naif0012.tls",
+        "imap_science_0001.tf",
+        "imap_sclk_0000.tsc",
+        "imap_wkcp.tf",
+        "imap_spin.bc",
+    ]
+
+    for file in test_dir.iterdir():
+        if file.name in kernels:
+            shutil.copy(file, tmp_path / file.name)
+
+    return tmp_path
+
+
+@pytest.fixture()
+def create_kernel_list(kernel_path):
+    """Create kernel lists."""
+    kernels = [str(file) for file in kernel_path.iterdir()]
+    ck_kernel = [
+        str(file) for file in kernel_path.iterdir() if file.name == "imap_spin.bc"
+    ]
+
+    return kernels, ck_kernel
+
+
+@pytest.mark.xfail(reason="Will fail unless kernels in pointing_frame/test_data.")
+def test_get_coverage(create_kernel_list):
+    """Tests get_coverage function."""
+    kernels, ck_kernel = create_kernel_list
+
+    with spice.KernelPool(kernels):
+        et_start, et_end, et_times = get_coverage(str(ck_kernel[0]))
+
+    assert et_start == 802008069.184905
+    assert et_end == 802094467.184905
+
+
+@pytest.mark.xfail(reason="Will fail unless kernels in pointing_frame/test_data.")
+def test_create_pointing_frame(monkeypatch, kernel_path, create_kernel_list):
+    """Tests create_pointing_frame function."""
+    monkeypatch.setenv("EFS_MOUNT_PATH", str(kernel_path))
+    _, ck_kernel = create_kernel_list
+    create_pointing_frame()
+
+    # After imap_dps.bc has been created.
+    kernels = [str(file) for file in kernel_path.iterdir()]
+
+    with spice.KernelPool(kernels):
```

Comment:
I would expect this to do a targeted furnish of the output from `create_pointing_frame()` and any other needed kernels (are there any?) rather than everything in the directory.

Reply:
I listed the kernels to furnish here, but I suspect this will change with your ensure spice code anyways.

    kernels = [
        "naif0012.tls",
        "imap_science_0001.tf",
        "imap_sclk_0000.tsc",
        "imap_wkcp.tf",
        "imap_spin.bc",
    ]